src/Controllers/Common/ContentController.cs

```diff
@@ -2293,4 +2194,106 @@ public IActionResult GetRevealIndex()
+                inventoryItemsToAdd.TryAdd(i.Key, 0);
+                inventoryItemsToAdd[i.Key] += i.Value;
+                itemsToSendBack.TryAdd(i.Key, 0);
+                itemsToSendBack[i.Key] += i.Value;
+            }
+            else if (itemService.IsGemBundle(i.Key, out int gemValue)) {
+                // get gem value
+                gemsToAdd += gemValue * i.Value;
+                itemsToSendBack.TryAdd(i.Key, 0);
+                itemsToSendBack[i.Key] += i.Value;
+            }
+            else if (itemService.IsCoinBundle(i.Key, out int coinValue)) {
+                // get coin value
+                coinsToAdd += coinValue * i.Value;
+                itemsToSendBack.TryAdd(i.Key, 0);
+                itemsToSendBack[i.Key] += i.Value;
+            }
+            else if (itemService.IsBundleItem(i.Key)) {
+                ItemData bundleItem = itemService.GetItem(i.Key);
+                // resolve items in the bundle
+                foreach (var reward in bundleItem.Relationship.Where(e => e.Type == "Bundle")) {
+                    int quantity = itemService.GetBulkItemQuantity(reward, i.Value);
+                    inventoryItemsToAdd.TryAdd(reward.ItemId, 0);
+                    inventoryItemsToAdd[reward.ItemId] += quantity;
+                    itemsToSendBack.TryAdd(reward.ItemId, 0);
+                    itemsToSendBack[reward.ItemId] += quantity;
+                }
+            }
+            else if (itemService.IsBoxItem(i.Key)) {
+                // open boxes individually
+                for (int j = 0; j < i.Value; j++) {
+                    itemService.OpenBox(i.Key, gender, out int itemId, out int quantity);
+                    inventoryItemsToAdd.TryAdd(itemId, 0);
+                    inventoryItemsToAdd[itemId] += quantity;
+                    itemsToSendBack.TryAdd(itemId, 0);
+                    itemsToSendBack[itemId] += quantity;
+                }
+            }
+            else {
+                // add item to inventory
+                inventoryItemsToAdd.TryAdd(i.Key, 0);
+                inventoryItemsToAdd[i.Key] += i.Value;
+                itemsToSendBack.TryAdd(i.Key, 0);
+                itemsToSendBack[i.Key] += i.Value;
+            }
+        }
+
+        // check if the user can afford the purchase
+        if (currency.GameCurrency - totalCoinCost < 0 && currency.CashCurrency - totalGemCost < 0) {
+            return new CommonInventoryResponse {
+                Success = false,
+                CommonInventoryIDs = new CommonInventoryResponseItem[0],
+                UserGameCurrency = achievementService.GetUserCurrency(viking)
+            };
+        }
+
+        // deduct the cost of the purchase
+        achievementService.AddAchievementPoints(viking, AchievementPointTypes.GameCurrency, -totalCoinCost + gemsToAdd);
+        achievementService.AddAchievementPoints(viking, AchievementPointTypes.CashCurrency, -totalGemCost + coinsToAdd);
```

Comment:
gems and coins are mixing here ... so when buy coins I get gems and vice versa

Reply:
Fixed